pkg/cortex/modules.go

```diff
@@ -419,7 +421,7 @@ var modules = map[moduleName]module{
 	},
 
 	Ingester: {
-		deps: []moduleName{Overrides, Store, Server},
+		deps: []moduleName{Overrides, Store, Ring, Server},
```

Comment:
Ingester already has access to the ring, doesn't need to initialize Ring component. (in Lifecycler.KVStore)

Reply:
In the ingester I need `ring.RingRead`, while the `Lifecycler` has a different interface (because they both cover different use cases on the same ring). This is the reason why I've initialized the ring here.

Do you mean refactoring `ring.New()` to accept `kv.Client` in input, so that we can reuse the `Lifecycler.KVStore`?

Reply:
I'm just pointing out that ingester is already reading and updating the ring via its lifecycler, and  it may be possible to use that (esp. if it is only to discover number of other ingesters), instead of introducing dependency on Ring component. Ring component watches the KV for updates constantly, which puts quite a load on the KV store. If all ingesters are going to do that as well, the load will undoubtedly increase.

Initializing two KV clients in memory will also cause issues when using memberlist-based KV store, and that will need to be handled somehow. (Eg. by using our suggestion -- both ring.New and Lifecycler using the same KVClient instance)

I'm not sure what the correct answer is here. I just wanted to bring it up as a potential issue, but if you have already considered alternatives and selected this, perhaps it is a correct answer.

Reply:
Thanks @pstibrany for your feedback. I've [commited a change](https://github.com/cortexproject/cortex/pull/1760/commits/24260ba9b20a0531068ce040c08a1b2a9c0be66f) to rollback the introduction of the `ReadRing` for the ingesters and count the number of healthy ingesters from the `Lifecycler`. May you re-review it, please?